datalad/distribution/create.py

```diff
@@ -10,78 +10,178 @@
-            repository/location, e.g. "Precious data on my laptop."""""),
+        description=dataset_description,
+        add_to_super=add_to_superdataset,
         no_annex=Parameter(
             args=("--no-annex",),
-            doc="""flag that if given a plain Git repository will be created
-            without any annex""",
+            doc="""if set, a plain Git repository will be created without any
+            annex""",
             action='store_false'),
         annex_version=Parameter(
             args=("--annex-version",),
-            doc="""select particular annex repository version.  The list of
-            supported versions depends on the available git-annex version""",
+            doc="""select a particular annex repository version. The
+            list of supported versions depends on the available git-annex
+            version. This should be left untouched, unless you know what
+            you are doing""",
             constraints=EnsureDType(int) | EnsureNone()),
         annex_backend=Parameter(
             args=("--annex-backend",),
+            constraints=EnsureStr() | EnsureNone(),
             # not listing choices here on purpose to avoid future bugs
             doc="""set default hashing backend used by the new dataset.
             For a list of supported backends see the git-annex
-            documentation""",
+            documentation. The default is optimized for maximum compatibility
+            of datasets across platforms (especially those with limited
+            path lengths)""",
             nargs=1),
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_init_opts=annex_init_opts,
     )
 
     @staticmethod
-    @datasetmethod(name='create', dataset_argname='loc')
+    @datasetmethod(name='create', dataset_argname='path')
     def __call__(
-            loc=None, description=None, no_annex=False, annex_version=None,
-            annex_backend='MD5E'):
-        if description and no_annex:
-            raise ValueError("Incompatible arguments: cannot specify description for "
-                             "annex repo and declaring no annex repo.")
-        if loc is None:
-            loc = os.curdir
-        elif isinstance(loc, Dataset):
-            loc = loc.path
-        if no_annex:
-            lgr.info("Creating a new git repo at %s", loc)
-            vcs = GitRepo(loc, url=None, create=True)
+            path=None,
+            description=None,
+            add_to_super=False,
+            no_annex=False,
+            annex_version=None,
+            annex_backend='MD5E',
+            git_opts=None,
+            annex_opts=None,
+            annex_init_opts=None):
+            # TODO: name? (technically not equal to path)
+
+        if path:
+            if isinstance(path, Dataset):
+                ds = path
+            else:
+                ds = Dataset(path)  # TODO: Is there a need to resolve path?
```

Comment:
I don't think there is a need to resolve a path. This is where things should be created, and both relative and absolute path are well-defined without the need or benefit of a reference dataset. Also as a bound function `create()` doesn't take a path, so the to-be-created dataset is the reference too.